### ISSUE TITLE
Detailed list doesn't work with docs:preview (only works with docs:dev)

### ISSUE THREAD

**Issue by Fil** [vuejs/vitepress#2485]
### Describe the bug

Hello, thank you so much for vitepress — we are using it for https://observablehq.com/plot/

Currently we see that the instant search **detailed list** feature works in `docs:dev` but does not work with `docs:preview` (nor in production). 

We are currently using v1.0.0-alpha.72, but I tried locally upgrading to v1.0.0-beta.1 and the issue is still here.

### Reproduction

Clone https://github.com/observablehq/plot
change vitepress version to v1.0.0-beta.1 in package.json
run yarn; yarn docs:build; yarn docs:preview
open the site and search “color”

screenshot: left: docs:dev; right: docs:preview

![vitepress-search-screen](https://github.com/vuejs/vitepress/assets/7001/72388a1e-2387-4ae6-8c55-505a97b4ac8d)

### Expected behavior

The detailed list should appear.

### System Info

```sh
System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1
    Memory: 63.64 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.1.0 - ~/.nvm/versions/node/v20.1.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.4 - ~/.nvm/versions/node/v20.1.0/bin/npm
  Browsers:
    Chrome: 114.0.5735.106
    Firefox: 113.0.2
    Safari Technology Preview: 16.4
  npmPackages:
    vitepress: ^1.0.0-beta.1 => 1.0.0-beta.1
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Somehow `render` function is missing from `comp` here

https://github.com/vuejs/vitepress/blob/d62e6f6dd68e1ac31654e0da6102915a5a3709f0/src/client/theme-default/components/VPLocalSearchBox.vue#L161-L162

**Comment by brc-dd** [comment]
I think we can fix this, but there is a change required at your side too. Your csv file paths need slight refactoring:

```diff
-  d3.csv("../data/aapl.csv", d3.autoType).then((data) => (aapl.value = data));
-  d3.csv("../data/goog.csv", d3.autoType).then((data) => (goog.value = data));
-  d3.csv("../data/sf-temperatures.csv", d3.autoType).then((data) => (sftemp.value = data));
+  d3.csv(withBase("/data/aapl.csv"), d3.autoType).then((data) => (aapl.value = data));
+  d3.csv(withBase("/data/goog.csv"), d3.autoType).then((data) => (goog.value = data));
+  d3.csv(withBase("/data/sf-temperatures.csv"), d3.autoType).then((data) => (sftemp.value = data));
```

`withBase` can be imported from `vitepress`.

**Comment by brc-dd** [comment]
fixed in 80e734d
